docs/usage/writing-logic-thunks.mdx

```diff
@@ -0,0 +1,463 @@
+
+Thunks are "one-shot" functions, with no sense of a lifecycle. They also cannot see other dispatched actions. So, they should not generally be used for initializing persistent connections like websockets, and you can't use them to respond to other actions.
+
+**Thunks are best used for complex synchronous logic, and simple to moderate async logic such as making a standard AJAX request and dispatching actions based on the request results.**
+
+## Redux Thunk Middleware
+
+Dispatching thunk functions requires that the [`redux-thunk` middleware](https://github.com/redujs/redux-thunk) has been added to the Redux store as part of its configuration.
+
+### Adding the Middleware
+
+The [Redux Toolkit `configureStore` API automatically adds the thunk middleware during store creation](../tutorials/fundamentals/part-8-modern-redux.md#using-configurestore), so it should typically be available with no extra configuration needed.
+
+If you need to add the thunk middleware to a store manually, that can be done by [passing the thunk middleware to `applyMiddleware()`](..//tutorials/fundamentals/part-6-async-logic.md#configuring-the-store) as part of the setup process.
+
+### How Does the Middleware Work?
+
+[Redux middleware are all written as a series of 3 nested functions](../tutorials/fundamentals/part-4-store.md#writing-custom-middleware):
+
+- The outer function receives a "store API" object with `{dispatch, getState}`
+- The middle function receives the `next` middleware in the chain (or the actual `store.dispatch` method)
+- The inner function will be called with each `action` as it's passed through the middleware chain
+
+It's important to note that middleware can be used to allow passing values that are _not_ action objects into `store.dispatch()`, as long as the middleware intercepts those values and does not let them reach the reducers.
+
+The actual implementation of the thunk middleware is very short - only about 10 lines. Here's the source, with additional added comments:
+
+```js title="Redux thunk middleware implementation, annotated"
+// standard middleware definition, with 3 nested functions:
+// 1) Accepts `{dispatch, getState}`
+// 2) Accepts `next`
+// 3) Accepts `action`
+const thunkMiddleware =
+  ({ dispatch, getState }) =>
+  next =>
+  action => {
+    // If the "action" is actually a function instead...
+    if (typeof action === 'function') {
+      // then call the function and pass `dispatch` and `getState` as arguments
+      return action(dispatch, getState)
+    }
+
+    // Otherwise, it's a normal action - send it onwards
+    return next(action)
+  }
+```
+
+In other words:
+
+- If you pass a function into `dispatch`, the thunk middleware sees that it's a function instead of an action object, intercepts it, and calls that function with `(dispatch, getState)` as its arguments
+- If it's a normal action object (or anything else), it's forwarded to the next middleware in the chain
+
+### Injecting Values Into Thunks
```

Comment:
I thought this title meant `dispatch(fetchData(value))` but this is not what this was about. Consider renaming to something like `Injecting Arguments into Thunks Via Middleware Config` or something a bit more specific? The explanation of why is really good here, just thinking a diff title might help?